security/security/src/main/java/io/helidon/security/DefaultAuditProvider.java

```diff
@@ -46,30 +66,40 @@ public static DefaultAuditProvider create(Config config) {
 
     private void audit(TracedAuditEvent event) {
         String tracingId = event.tracingId();
+        Level level;
+
         switch (event.severity()) {
         case FAILURE:
+            level = failureLevel;
+            break;
         case SUCCESS:
+            level = successLevel;
+            break;
         case INFO:
-            //trace info
-            logEvent(tracingId, event, Level.FINEST);
+            level = infoLevel;
             break;
         case WARN:
-            //warning - something is not right, so let's log it
-            logEvent(tracingId, event, Level.WARNING);
+            level = warnLevel;
             break;
         case ERROR:
-            //error - definitely a problem, log as severe
-            logEvent(tracingId, event, Level.SEVERE);
+            level = errorLevel;
             break;
         case AUDIT_FAILURE:
         default:
             //audit failure - something wrong with auditing mechanism...
-            logEvent(tracingId, event, Level.SEVERE);
+            level = auditFailureLevel;
             break;
         }
+
```

Comment:
Assuming you remove the allocation of the `tracingId` variable above, then you can perform the `if (!auditLogger.isLoggable(level))` check here instead of (or in addition to) in the `logEvent` method.